File: flink-sample-scala/src/main/scala/util/SensorTimeAssigner.scala
package util

import org.apache.flink.streaming.api.functions.timestamps.BoundedOutOfOrdernessTimestampExtractor
import org.apache.flink.streaming.api.windowing.time.Time

class SensorTimeAssigner extends BoundedOutOfOrdernessTimestampExtractor[SensorReading](Time.seconds(5)) {
  override def extractTimestamp(t: SensorReading): Long = t.timestamp
}

File: flink-datastream-1.7.1/src/main/java/timestamp/SensorTimeAssigner.java
package timestamp;

import org.apache.flink.streaming.api.functions.timestamps.BoundedOutOfOrdernessTimestampExtractor;
import org.apache.flink.streaming.api.windowing.time.Time;
import util.SensorReading;

public class SensorTimeAssigner extends BoundedOutOfOrdernessTimestampExtractor<SensorReading> {
    public SensorTimeAssigner() {
        super(Time.seconds(5));
    }
    @Override
    public long extractTimestamp(SensorReading sensorReading) {
        return sensorReading.timestamp;
    }
}
